### PR TITLE
Open the new window with WebContents instead of initial url

### PR DESCRIPTION
**Pull request by mshin-wolvic**
This patch changes the opening window's parameter as WebContents instead of initial url so that keeps the connection of a parent window and a child window(=the popup). Also the patch changes a management of WebContents lifecycle to fix the leakage of it.

### CODE REVIEW

File: wolvic/browser/wolvic_web_contents_delegate.cc [hunk 1]
```diff
@@ -38,21 +37,24 @@ void WolvicWebContentsDelegate::AddNewContents(
...
+  // |new_contents| ownership has been passed to java, and will retake it
+  // in WolvicWebContents when the new tab is created asynchronously.
+  new_contents.release();
+}
+
+bool WolvicWebContentsDelegate::ShouldResumeRequestsForCreatedWindow() {
+  // Always return false here since we need to defer loading the created window
+  // until after we have attached a new delegate to the new webcontents (which
+  // happens asynchronously).
+  return false;
```
Comment (svillar) [inline wolvic/browser/wolvic_web_contents_delegate.cc:57]: This is awesome. That's indeed what I wanted to do when I implemented the current solution.

File: wolvic/java/org/chromium/wolvic/Tab.java [hunk 2]
```diff
@@ -44,20 +44,33 @@ public WebContents createWebContents(boolean is_off_the_record) {
         return TabJni.get().createWebContents(is_off_the_record);
     }
 
+    private void setWebContents(WebContents mWebContents) {
+        TabJni.get().setWebContents(mWebContents);
+    }
+
+    private void destroyWebContents(WebContents webContents) {
+        TabJni.get().destroyWebContents(webContents);
+    }
```
Comment (svillar) [inline wolvic/java/org/chromium/wolvic/Tab.java:53]: I am not sure about these two names, I'll explain it in the next comment

File: wolvic/java/org/chromium/wolvic/Tab.java [hunk 3]
```diff
@@ -44,20 +44,33 @@ public WebContents createWebContents(boolean is_off_the_record) {
...
                 IntentRequestTracker.createFromActivity(ContextUtils.activityFromContext(context)));
 
         mCompositorView = new TabCompositorView(context);
         mCompositorView.onNativeLibraryLoaded(mWindowAndroid);
         mWindowAndroid.setAnimationPlaceholderView(mCompositorView);
 
-        mWebContents = createWebContents(is_off_the_record);
+        if (webContents != null) {
+            mWebContents = webContents;
+            setWebContents(mWebContents);
```
Comment (svillar) [inline wolvic/java/org/chromium/wolvic/Tab.java]: This is the reason why I dislike `setWebContents`. Typically a setter does precisely what the previous line does, i.e. assign some object to an attribute. I think we should use a more descriptive name, perhaps `registerWolvicContents` although I don't like it too much either. Perhaps you are aware of a better naming for this kind of things in Chromium.

File: wolvic/java/org/chromium/wolvic/Tab.java [hunk 4]
```diff
@@ -73,6 +86,18 @@ public Tab(@NonNull Context context, boolean is_off_the_record) {
...
+        mWindowAndroid = null;
+        mContentView = null;
+        mNavigationController = null;;
+        mCompositorView = null;;
+
+        if (!mWebContents.isDestroyed()) {
+            destroyWebContents(mWebContents);
+        }
+        mWebContents = null;;
+    }
```
Comment (svillar) [inline wolvic/java/org/chromium/wolvic/Tab.java:99]: Several `;;` in this method.

**Review by svillar — CHANGES_REQUESTED** [review]
This is great, I just have some minor comments.

File: wolvic/java/org/chromium/wolvic/Tab.java (hunk 3 above)
Comment (mshin-wolvic) [reply wolvic/java/org/chromium/wolvic/Tab.java]: I renamed them that is a little more meaningful.  Thank you for giving me the chance to rethink the name.

File: wolvic/java/org/chromium/wolvic/Tab.java (hunk 4 above)
Comment (mshin-wolvic) [reply wolvic/java/org/chromium/wolvic/Tab.java:99]: Done

File: wolvic/java/org/chromium/wolvic/Tab.java (hunk 2 above)
Comment (mshin-wolvic) [reply wolvic/java/org/chromium/wolvic/Tab.java:53]: Done. Please review the new name.

**Review by svillar — APPROVED** [review]
Great! Let's merge this

**Pull request merged**